allennlp/training/metrics/tqdm_metrics_reporter.py

```diff
@@ -0,0 +1,20 @@
+from typing import Dict, Iterable
+
+
+class TqdmMetricsReporter(Iterable):
```

Comment:
I don't think this abstraction buys us anything.  It moves one one-line method off of `trainer` and onto this, while adding a whole new class and complicating other parts of the trainer code (because you need to know about and instantiate a new object to wrap the `tqdm`).

And in general, every time you add a class to the codebase, you add complexity - docs that need to be maintained, more things to know about when writing code, and more things to modify if we want to refactor the code later.  Sometimes that complexity is definitely worth it.  But for really simple things, like in the two abstractions you've added here, the cost in additional complexity isn't worth the benefit that you get from using the abstraction.  We need to be really careful to not just add abstractions for abstraction's sake.